src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,88 @@
+package duke.command;
+
+import duke.Duke;
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Represents a user command that adds a task to the task list.
+ * This includes user input of todo, deadline, and event.
+ */
+public class AddCommand extends Command {
+    private String userInput;
+
+    public AddCommand(String userInput) {
+        this.userInput = userInput;
+    }
+
+    // checks if the program should exit.
+    public boolean isExit() {
+        return false;
+    }
+
+    /**
+     * Checks for the type of task added by the user based on the user input, and
+     * adds the appropriate type of task to the task list.
+     *
+     * @param tasks List of <code>Task</code> objects.
+     * @param ui Ui object created by Duke.
+     * @param storage Storage object created by Duke.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
```

Comment:
Perhaps extract the code blocks within each `if (t.isSomething())` out to separate functions? Otherwise as more task types add on, this method will get very long.